### ISSUE TITLE
[Bug] Callbacks are not being called after requesting permissions

### ISSUE THREAD

**Issue by vrazn** [Jumio/mobile-react#32]
Hi!

We've encountered an interesting bug while testing our integration:
Android SDK does not invoke `EventResult` or `EventError` events if user is being asked for camera permissions at the beginning of the flow.

Tested on: SDK v4.1.0 & 4.2.0

Steps to reproduce:
1) Make a fresh install of the demo app (or just set the camera permissions on Android to 'always ask' in system settings). This is required to trigger the permissions dialog.
2) Turn on the debugger
3) Input the token & start the flow
4) Accept the camera permission
5) Proceed to the document capture screen
6) Either go back to the main page of the app or finish the flow & check your console. It will be empty.
7) Repeat steps 3-6. The message will be displayed this time.

Expected result:
1) Upon going back to the main screen the `EventError` should be triggered, specifying that the end user has cancelled the flow.
2) On successful upload the `EventResult` should be triggered & data should be available.

Potential cause:

```
@ReactMethod
    fun start() {
        try {
             // This will always return false if the user doesn't have camera permissions upon pressing the button
            val sdkStarted = checkPermissionsAndStart()
            if (sdkStarted) {
                // hence we never attach the listener
                reactContext.addActivityEventListener(mActivityEventListener)
            }
        } catch (e: Exception) {
            showErrorMessage("Error starting the Jumio SDK: " + e.localizedMessage)
        }
    }
```

**Comment by vrazn** [comment]
@bratapfelmus Sorry, I guess this behavior only happens on the v4.1.0 of the SDK.
We had to downgrade because the app re-renders when we try to use v4.2.0 

 Let me create a reproducible example for you.

**Comment by bratapfelmus** [comment]
@vrazn 
please do and include it in your support request. 
Please also specify what exactly you mean when saying "the app re-renders" for 4.2.0

**Comment by bratapfelmus** [comment]
Hello @vrazn 
Please contact Jumio support with these findings.

**Comment by BernadetteTh** [comment]
Resolved as of version 4.3.1
